### PR TITLE
[fix](fe) Preserve delete fallback failure cause

### PR DESCRIPTION
**Pull request by wenzhenghu**
### What problem does this PR solve?

Related PR: https://github.com/apache/doris/pull/65697

Problem Summary: Preserve both the initial predicate-check failure and the real fallback execution failure when `DeleteFromUsing` fallback also fails.

### Release note

None

### Check List (For Author)

- Test: FE Unit Test
  - `./run-fe-ut.sh --run org.apache.doris.nereids.trees.plans.commands.DeleteFromCommandTest`
- Behavior changed: Yes (delete fallback errors now expose both causes)
- Does this need documentation: No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteFromCommand.java [hunk 1]
```diff
@@ -168,11 +168,12 @@ public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
             }
         } catch (Exception e) {
             try {
+                // Fall back to the write-based delete path when predicate delete validation fails.
                 new DeleteFromUsingCommand(nameParts, tableAlias, isTempPart, partitions,
                         logicalQuery, Optional.empty()).run(ctx, executor);
                 return;
             } catch (Exception e2) {
-                throw e;
+                throw buildDeleteFallbackException(e, e2);
```
Comment (Ryan19929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteFromCommand.java:176]: [P2] Preserve the fallback exception type and MySQL error code

When DeleteFromUsingCommand fails with a retryable cloud insert DdlException (for example ERR_FAILED_WHEN_INSERT, code 5025), wrapping it in Nereids AnalysisException makes StmtExecutor enter the generic catch (Exception) branch. Although the combined message still contains the retry marker, the typed UserException and its MySQL error code are lost; after retry exhaustion the client can receive generic 1105 instead of 5025. The direct DeleteFromUsingCommand path and the MOW-light-delete-disabled path preserve the original exception.

Please keep e2 as the primary throwable (for example, attach e as suppressed and rethrow e2), or otherwise explicitly preserve its UserException/error-code semantics while combining diagnostics. Also add a test through this catch and the StmtExecutor boundary: the current helper-only tests would still pass if this line were reverted to throw e.

**Review by Ryan19929 — CHANGES_REQUESTED** [review]
I found one correctness issue in the exception propagation path. The diagnostic goal is valid, but the fallback exception typed client error must remain primary. The current helper-only tests do not cover this command/StmtExecutor boundary.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteFromCommand.java (hunk 1 above)
Comment (wenzhenghu) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteFromCommand.java:176]: Thanks for the suggestion. We checked the corresponding upstream PR `apache/doris#65697` and aligned this change with the upstream implementation. That PR has already been merged into `master`, and it did not adopt the same additional fix mentioned in this P2 comment. Since the goal of this change is to backport the upstream fix and keep behavior consistent with upstream while minimizing maintenance divergence, we prefer to keep `HYDCP/hy-doris#69` aligned with the community implementation and not introduce an extra difference in this PR.

If there is a follow-up change in upstream for this point, we can sync it later as well.
